app/services/DomainService.scala

```diff
@@ -80,22 +81,33 @@ class DomainService {
         .map(_.map(pw => toDomain(pw.paymentAndWithdrawal))).getOrElse(Seq.empty))
   }
 
-
   def toDomainDetail(declaration: DeclarationDetail): domain.Declaration = {
     Declaration(declaration.declarationID,
       declaration.importerEORINumber,
       declaration.declarantEORINumber,
       declaration.declarantReference,
       declaration.postingDate,
       declaration.amount,
-      declaration.taxGroups.map(container => toDomain(container.taxGroup)))
+      declaration.taxGroups.map(container => toDomainTaxGroup(container.taxGroup)))
```

Comment:
Could you please write tests for this change i.e. for toDomainDetail method? If it takes time then leave it.